mavlink_vehicles/mavlink_vehicles.cc

```diff
@@ -390,12 +396,34 @@ void msghandler::handle(mav_vehicle &mav, const mavlink_message_t *msg)
         global_mission_item.timestamp = std::chrono::system_clock::now();
         global_mission_item.is_new = true;
 
-        // Retrieve current mission item data converting from
-        // global-relative-alt to global
+        // Mission item received
+        print_verbose("Mission item received: %d\n", mission_item.seq);
+
+        // Store mission item on list
+        if (mav.mission_items_list_received.size() == mission_item.seq) {
+            mav.mission_items_list_received.push_back(global_mission_item);
+        }
+
+        // Request next mission item
+        if (mav.mission_items_list_received.size() <
+            mav.mission_items_list_received_size) {
```

Comment:
I thought you were going to change those names to avoid confusion. It's a minor issue though.